FF1Lib/Credits.cs

```diff
@@ -348,19 +377,69 @@ private void SetupStoryPages(MT19337 rng)
 				FF1Text.TextToBytes("        ",     true, FF1Text.Delimiter.Empty), Blob.FromHex("101003B26004919900"),
 			};
 
-			pages.Add(FF1Text.TextToStory(VictoryMessages[rng.Between(0, VictoryMessages.Count - 1)]));
+			var victory = FF1Text.TextToStory(VictoryMessages[rng.Between(0, VictoryMessages.Count - 1)]); 
+
+			//pages.Add(FF1Text.TextToStory(VictoryMessages[rng.Between(0, VictoryMessages.Count - 1)]));
+			pages.Add(victory);
+
+
+			// this needs to go right before the stats pages; this address is used
+			// in StatsTracking.cs
+			StatsPageAddress = StoryTextAddress + pages.Count * 2;
 			pages.Add(Blob.Concat(movementStats));
 			pages.Add(Blob.Concat(battleResults));
 			pages.Add(Blob.Concat(combatStats));
 			ThankYous.ForEach(page => pages.Add(FF1Text.TextToStory(page)));
 
-			Blob storyText = PackageTextBlob(pages, 0xA800);
-			System.Console.WriteLine($"Credits Saved. 0x{storyText.Length:X2}B of 0x500B used.");
-			System.Diagnostics.Debug.Assert(storyText.Length <= 0x0500, $"Story text {storyText.Length:X2} too large!");
 
-			Put(0x36800, storyText);
-			Data[0x36E00] = (byte)(BridgeStory.Count);
-			Data[0x36E01] = (byte)(pages.Count - 1);
+			Blob storyText = PackageTextBlob(pages, StoryTextAddress);
+
+			// bounds check: 0x885 bytes until the next important thing
+			if (storyText.Length > 0x885)
+			{
+				await Progress($"WARNING: custom credit text is too long by {(storyText.Length - 0x885):X2} bytes. Reverting to default.");
+				pages.Clear();
+				DefaultStory.ForEach(page => pages.Add(FF1Text.TextToStory(page)));
+				pages.Add(victory);
+				StatsPageAddress = StoryTextAddress + pages.Count * 2;
+				pages.Add(Blob.Concat(movementStats));
+				pages.Add(Blob.Concat(battleResults));
+				pages.Add(Blob.Concat(combatStats));
+				ThankYous.ForEach(page => pages.Add(FF1Text.TextToStory(page)));
+
+				storyText = PackageTextBlob(pages, StoryTextAddress);	
+			}
+
+			System.Console.WriteLine($"Credits Saved. 0x{storyText.Length:X2} Bytes used.");
+			// System.Diagnostics.Debug.Assert(storyText.Length <= 0x0500, $"Story text {storyText.Length:X2} too large!");
+
+
+			// move some stuff to make room for longer bridge story from resource packs.
+			// To complicate matters, StatsTrackingHacks moves some of these instructions to a new bank,
+			// but still tries to point to bank 0x0D. That bug wasn't noticed because it just pointed to
+			// empty ROM space in 0x1E, giving an extra page of credits at the end.
+			// This would be correct if things stayed in 0x0D:
+			// PutInBank(0x0D,0xB812, Blob.FromHex(AddressToHex(BridgeLastPageAddress)));
+			// We take advantage of an unnecessary clearing of the story_page to load the value instead
+			Data[0x0D * 0x4000 - 0x8000 + 0xB89F] = (byte)BridgeStory.Count;

```

Comment:
there's isn't any difference between assigning to Data[] and using PutInBank, but PutInBank take care of computing the offset and is easier to read, so these should be changed to PutInBank

Reply:
For a single byte, it's either this, or create a new Blob from a new byte[] {mybyte}. Either way it's ugly. Maybe one day PutInBank() should have overloads for single bytes and ushorts